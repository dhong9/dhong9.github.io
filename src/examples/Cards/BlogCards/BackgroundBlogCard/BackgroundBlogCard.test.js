// React testing libraries
import React from "react";
import renderer from "react-test-renderer";

// @mui material components
import { ThemeProvider } from "@mui/material/styles";
import theme from "assets/theme";

// Component to test
import BackgroundBlogCard from "examples/Cards/BlogCards/BackgroundBlogCard";

let realUseContext;
let useContextMock;
// Setup mock
beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = jest.fn(() => "Test Value");
});
// Cleanup mock
afterEach(() => {
    React.useContext = realUseContext;
});

// Mocks
jest.mock("react-router-dom", () => ({
  Link: jest.fn(({ to, children }) => <a href={to}>{children}</a>),
}));

describe("BackgroundBlogCard", () => {
  it("renders", () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <BackgroundBlogCard
          title="Boeing 747"
          description="Queen of the skies"
          image="https://jooinn.com/images1280_/boeing-747.jpg"
          action={{ type: "internal", route: "/", label: "Read more..." }}
        />
      </ThemeProvider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
