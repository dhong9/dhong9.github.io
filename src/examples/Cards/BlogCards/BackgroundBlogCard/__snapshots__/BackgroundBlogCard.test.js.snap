// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundBlogCard renders external type 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-tj95r-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiBox-root css-hwgoop"
  >
    <div
      className="MuiBox-root css-fyv4a6"
    >
      <h2
        className="MuiTypography-root MuiTypography-h2 css-1qu2wyd-MuiTypography-root"
      >
        Boeing 747
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body2 css-1fvfs31-MuiTypography-root"
      >
        Queen of the skies
      </p>
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone MuiTypography-root MuiTypography-body2 css-jxqcqk-MuiTypography-root-MuiLink-root-MuiTypography-root"
        href="/"
        onBlur={[Function]}
        onFocus={[Function]}
        rel="noreferrer"
        target="_blank"
      >
        Read more...
        <span
          aria-hidden={true}
          className="material-icons-round notranslate MuiIcon-root MuiIcon-fontSizeInherit css-fnit94-MuiIcon-root"
        >
          arrow_forward
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`BackgroundBlogCard renders internal type 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-tj95r-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiBox-root css-hwgoop"
  >
    <div
      className="MuiBox-root css-fyv4a6"
    >
      <h2
        className="MuiTypography-root MuiTypography-h2 css-1qu2wyd-MuiTypography-root"
      >
        Boeing 747
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body2 css-1fvfs31-MuiTypography-root"
      >
        Queen of the skies
      </p>
      <a
        href="/"
      >
        Read more...
        <span
          aria-hidden={true}
          className="material-icons-round notranslate MuiIcon-root MuiIcon-fontSizeInherit css-fnit94-MuiIcon-root"
        >
          arrow_forward
        </span>
      </a>
    </div>
  </div>
</div>
`;
